Tests/SourceKitLSPTests/CodeActionTests.swift

```diff
@@ -425,4 +425,54 @@ final class CodeActionTests: XCTestCase {
 
     try await fulfillmentOfOrThrow([editReceived])
   }
+
+  func testCodeActionForFixItsProducedBySwiftSyntax() async throws {
+    let ws = try await MultiFileTestWorkspace(files: [
+      "test.swift": "protocol Multi ident {}",
+      "compile_commands.json": "[]"
+    ])
+
+    let (uri, _) = try ws.openDocument("test.swift")
+
+    let report = try await ws.testClient.send(DocumentDiagnosticsRequest(textDocument: TextDocumentIdentifier(uri)))
+    guard case .full(let fullReport) = report else {
+      XCTFail("Expected full diagnostics report")
+      return
+    }
+
+    XCTAssertEqual(fullReport.items.count, 1)
+    let diagnostic = try XCTUnwrap(fullReport.items.first)
+    guard let codeActions = diagnostic.codeActions else {
+      XCTFail("Expected non-empty codeActions")
+      return
+    }
+
+    let expectedCodeActions = [
+      CodeAction(
+        title: "Replace \'Multi \' with \'Multiident \'...", 
+        kind: .quickFix, 
+        edit: WorkspaceEdit(
+          changes: [
+            uri : [
+              TextEdit(range: Position(line: 0, utf16index: 9)..<Position(line: 0, utf16index: 15), newText: "Multiident "),
```

Comment:
Could you use location markers here instead of hardcoded line-column offsets. We aren’t very good about it in the codebase yet as I only introduced the concept of location markers in the last couple of weeks but it’s generally proffered because of the same reasons that I mentioned in your swift-syntax PR.

Here’s an example of how to use the location markers.

https://github.com/apple/sourcekit-lsp/blob/9f7ff0891a115f1ae056348744abffd0452806e5/Tests/SourceKitLSPTests/LocalClangTests.swift#L246-L264

Reply:
got it, it makes sense to use location markers instead. i'll update the code.